src/main/java/htsjdk/samtools/reference/FastaReferenceWriter.java

```diff
@@ -0,0 +1,758 @@
+                                final boolean makeDictOutput)
+            throws IOException {
+        this(ValidationUtils.nonNull(fastaFile, "the output fasta-file cannot be null"),
+                basesPerLine,
+                defaultFaiFile(makeFaiOutput, fastaFile),
+                defaultDictFile(makeDictOutput, fastaFile));
+    }
+
+    /**
+     * Creates a reference FASTA file writer.
+     * <p>
+     * The default bases-per-line is set to {@link #DEFAULT_BASES_PER_LINE}.
+     * </p>
+     * <p>
+     * You can specify a specific path for the index and dictionary file. If either set to {@code null} such
+     * a file won't be generated.
+     * </p>
+     *
+     * @param fastaFile the output fasta file path.
+     * @param indexFile the path of the index file, if requested, {@code null} if none should be generated.
+     * @param dictFile  the path of the dictFile, if requested, {@code null} if nono should be generated.
+     * @throws IllegalArgumentException if {@code fastaFile} is {@code null}.
+     * @throws IOException              if such exception is thrown when accessing the output path resources.
+     */
+    public FastaReferenceWriter(final Path fastaFile, final Path indexFile, final Path dictFile)
+            throws IOException {
+        this(fastaFile, DEFAULT_BASES_PER_LINE, indexFile, dictFile);
+    }
+
+    /**
+     * Creates a reference FASTA file writer.
+     * <p>
+     * You can specify a specific path for the index and dictionary file. If either set to {@code null} such
+     * a file won't be generated.
+     * </p>
+     *
+     * @param fastaFile the output fasta file path.
+     * @param indexFile the path of the index file, if requested, {@code null} if none should be generated.
+     * @param dictFile  the path of the dictFile, if requested, {@code null} if nono should be generated.
+     * @throws IllegalArgumentException if {@code fastaFile} is {@code null} or {@code basesPerLine} is 0 or negative.
+     * @throws IOException              if such exception is thrown when accessing the output path resources.
+     */
+    public FastaReferenceWriter(final Path fastaFile, final int basesPerLine, final Path indexFile, final Path dictFile)
+            throws IOException {
+        // This code is a slight repeat of {@link #FastaReferenceWriter(OutputStream,int,OutputStream,OutputStream)
+        // for the sake of avoiding creating output if basesPerLine is invalid.
+        this.defaultBasePerLine = checkBasesPerLine(basesPerLine);
+
+        this.fastaStream = new CountingOutputStream(Files.newOutputStream(fastaFile));
+        this.indexWriter = indexFile == null ? NullWriter.NULL_WRITER : new OutputStreamWriter(Files.newOutputStream(indexFile), CHARSET);
+        final BufferedWriter dictWriter = new BufferedWriter(dictFile == null ? NullWriter.NULL_WRITER : new OutputStreamWriter(Files.newOutputStream(dictFile), CHARSET));
```

Comment:
the issue is that we want to check the basesPerLine before the OutputFile is created...and as the comment (in the other constructor says) this is hard to do without code duplication....

Everything else has to be in the constructor since it's final.

Reply:
nm. changed the signature and I think it works now.